src/other-reprs.md

```diff
@@ -112,26 +112,43 @@ assert_eq!(16, size_of::<MyReprOption<&u16>>());
 
 This optimization still applies to fieldless enums with an explicit `repr(u*)`, `repr(i*)`, or `repr(C)`.
 
-## repr(packed)
+## repr(packed(n))
 
-`repr(packed)` forces Rust to strip any padding, and only align the type to a
-byte. This may improve the memory footprint, but will likely have other negative
-side-effects.
+`repr(packed(n))` forces the Rust compiler to strip any padding between fields, and align the type, hence,
+the first field, at least to a [`n`-byte boundary][rust-reference-type-layout] (where `n` is a power of two).
+`repr(packed)` is equivalent to `repr(packed(1))`.
+
+This `repr`-attribute is a modifier on `repr(C)` and `repr(Rust)`, but is incompatible with `align(n)`. It may improve
+the memory footprint and is useful in scenarios, where you need exact control over the memory layout
+(such as constructing network packets).
 
 In particular, most architectures *strongly* prefer values to be aligned. This
```

Comment:
The "In particular" here is no longer clear, as it is now separated from the original "likely have other negative side-effects".  Perhaps the start of this paragraph can be rewritten to include the original warning?

Also, is there a reason to remove the comment about the compiler using shifts and masks to deal with an unaligned load?  That seems like it is useful to highlight if indeed that's how it works, since that highlights the performance penalties of using packed structures.

Reply:
In my opinion the old version sounded like it includes too much in-depth details. But you are right. I work on a better formulation.